pkg/provisioner/ironic/ironic.go

```diff
@@ -396,6 +397,13 @@ func (p *ironicProvisioner) ValidateManagementAccess(credentialsChanged, force b
 	if ironicNode == nil {
 		p.log.Info("registering host in ironic")
 
+		if p.host.Spec.BootMode == metal3v1alpha1.UEFISecureBoot && !p.bmcAccess.SupportsSecureBoot() {
+			msg := fmt.Sprintf("BMC driver %s does not support secure boot", p.bmcAccess.Type())
```

Comment:
It'd be nice to have a test for this error path - similar to the existing ones in https://github.com/metal3-io/baremetal-operator/blob/master/pkg/provisioner/ironic/validatemanagementaccess_test.go

Reply:
https://github.com/metal3-io/baremetal-operator/pull/831